djreact/djangoapps/trips/models.py

```diff
@@ -0,0 +1,154 @@
+import json
+from datetime import datetime
+
+from django.db import models
+from django.db.models import permalink
+
+
+class Host(models.Model):
+    """
+    Trip host model.
+
+    This model contains the information for the trip hosts who are organizing
+    trips.
+    """
+    name = models.CharField(max_length=30)
+    slug = models.SlugField(max_length=50, null=True, blank=True)
+    description = models.TextField(null=True, blank=True)
+    verified = models.BooleanField(default=False)
+
+
+class Location(models.Model):
+    """
+    Trip location model
+
+    This model contains information about trip location with respect to
+    coordinates. We will use coordinates to draw google map.
+    """
+    name = models.CharField(max_length=30)
+    slug = models.SlugField(max_length=50, null=True, blank=True)
+    coordinates = models.CharField(max_length=40, null=True, blank=True)
+
+
+class Activity(models.Model):
+    """
+    Trip activity model
+
+    This model contains what kind of activities a trip will offer.
+    """
+    name = models.CharField(max_length=30)
+    slug = models.SlugField(max_length=50, null=True, blank=True)
+
+
+class Facility(models.Model):
+    """
+    Trip Facility model
+
+    This model contains information all the available facilities that can be
+    provided in a trip.
+    """
+    name = models.CharField(max_length=30)
+    slug = models.SlugField(max_length=50, null=True, blank=True)
+
+
+class Trip(models.Model):
+    """
+    Trip model
+
+    This model contains the main information that will be presented to
+    end users.
+    """
+    objects = models.Manager()
+
+    name = models.CharField("Title", max_length=500, null=True, blank=True)
+    slug = models.SlugField(max_length=50, null=True, blank=True)
+    description = models.TextField("Description", null=True, blank=True)
+    # meta includes tinyurl, poster
+    _metadata = models.TextField(default='{}', null=True, blank=True)
+
+    duration = models.SmallIntegerField(default=0, null=True, blank=True)
+
+    starting_location = models.ForeignKey(Location)
```

Comment:
Why are we marking the starting_location as FK? Shouldn't it be a 1-1 relation, given that all trips are supposed to go from a single, decisive point?

Reply:
Good point.